From: John Johnansen <jjohansen@suse.de>
Subject: apparmor_file_mmap function parameters mismatch
Patch-mainline: no
References: 328423

The apparmor_file_mmap routine has 2 more parameters than the 2.6.22
security function.  The 2 extra parameters were introduced in 2.6.23 and
the changes leaked back into the 2.6.22 version of apparmor.  AppArmor
does not currently use these parameteres.

static int apparmor_file_mmap(struct file *file, unsigned long reqprot,
                              unsigned long prot, unsigned long flags,
                              unsigned long addr, unsigned long addr_only)

where the 2.6.22 security_file_mmap has a prototype of
int (*file_mmap) (struct file * file,
                  unsigned long reqprot,
                  unsigned long prot, unsigned long flags);

Signed-off-by: John Johansen <jjohansen@suse.de>


Index: linux-2.6.22/security/apparmor/lsm.c
=====================================================================
--- linux-2.6.22.orig/security/apparmor/lsm.c
+++ linux-2.6.22/security/apparmor/lsm.c
@@ -599,8 +599,7 @@ static inline int aa_mmap(struct file *f
 }
 
 static int apparmor_file_mmap(struct file *file, unsigned long reqprot,
-			      unsigned long prot, unsigned long flags,
-			      unsigned long addr, unsigned long addr_only)
+			      unsigned long prot, unsigned long flags)
 {
 	return aa_mmap(file, "file_mmap", prot, flags);
 }

